numba_cuda/numba/cuda/cudadrv/nvrtc.py

```diff
@@ -251,6 +251,10 @@ def compile(src, name, cc, ltoir=False):
     nvrtc = NVRTC()
     program = nvrtc.create_program(src, name)
 
+    if nvrtc.get_version() < (12, 0):
+        # maximum supported target architecture for 11.x is 8.0
+        cc = min(cc, (8, 0))
```

Comment:
CUDA 11.8 supports Hopper, arch 9.0.
```suggestion
        # Maximum supported target architecture for 11.x is cc 9.0 in CUDA 11.8
        cc = min(cc, (9, 0))
```

Reply:
Do we need a full range of support here? Do we need to account for CUDA 12.8 introducing Blackwell support?

Reply:
> Do we need a full matrix of arch/CUDA version support here? Do we need to account for CUDA 12.8 introducing Blackwell support?

You can target older architecture and the compiled binary would still work on newer hardware. In CTK 11's case this means you can only target at maximum for Hopper albeit you are running on Blackwell hardware. Although a warning saying that you may not fully exploit the machine's hardware capability could be helpful?

Reply:
> But nvrtc is still missing "sm_90" and "compute_90" option?
That seems like a bug in the docs. CUDA 11.8 should have full support for Hopper (sm90).

Reply:
> Do we need a full matrix of arch/CUDA version support here?

Sorry, my intent wasn't clear. What I meant was, do we need to spell out arch support for every CUDA version? Or is there some other reason why 11.8 is special?

```python
if nvrtc.get_version() < (11, 0):
    raise RuntimeError("Unsupported CUDA version. CUDA 11.0 or higher is required.")
elif nvrtc.get_version() < (11, 1):
    # Maximum supported target architecture for CUDA 11.0 is cc 8.0
    cc = min(cc, (8, 0))
elif nvrtc.get_version() < (11, 5):
    # Maximum supported target architecture for CUDA 11.1-11.4 is cc 8.6
    cc = min(cc, (8, 6))
elif nvrtc.get_version() < (11, 8):
    # Maximum supported target architecture for CUDA 11.5-11.7.1 is cc 8.7
    cc = min(cc, (8, 7))
elif nvrtc.get_version() < (12, 8):
    # Maximum supported target architecture for CUDA 11.8-12.6 is cc 9.0
    cc = min(cc, (9, 0))
```

But really, we already have this whole table written out in numba-cuda here, and we should use that instead of hardcoding something new: https://github.com/NVIDIA/numba-cuda/blob/fb7c1a2e2c0542327b251128b81f4c2b432ac7c2/numba_cuda/numba/cuda/cudadrv/nvvm.py#L337-L354

Reply:
> do we need to spell out arch support for every CUDA version? Or is there some other reason why 11.8 is special?
A full matrix would be good. The patch I provided was only poking at the CI error. And we can reuse the table you proposed. I was also hoping `cuda.core` rtc model can capture something like this nicely. cc @leofang

Q: What happens if the device CC is lower than the minimum supported CC by the CTK? Should we just err?

Reply:
> Q: What happens if the device CC is lower than the minimum supported CC by the CTK? Should we just err?

Yes. I don't know the error classes that might exist in numba/numba-cuda, but perhaps there's a good one that indicates "unsupported." It should probably inherit from `RuntimeError`, or maybe we just use `RuntimeError.`